xdsl/backend/riscv/register_allocation.py

```diff
@@ -24,6 +25,95 @@ def allocate_registers(self, module: ModuleOp) -> None:
         raise NotImplementedError()
 
 
+class RegisterAllocatorLivenessBlockNaive(RegisterAllocator):
+    idx: int
+
+    def __init__(self) -> None:
+        self.idx = 0
+
+        """
+        Assume that all the registers are available except the ones explicitly reserved
+        by the default RISCV ABI
+        """
+        self.reserved_registers = set(["zero", "sp", "gp", "tp", "fp", "s0"])
+
+        self.register_sets = {
+            RegisterType: [
+                reg
+                for reg in Register.RV32I_INDEX_BY_NAME
+                if reg not in self.reserved_registers
+            ],
+            FloatRegisterType: list(Register.RV32F_INDEX_BY_NAME.keys()),
+        }
+
+    @staticmethod
+    def _is_allocated(reg: SSAValue) -> bool:
+        return (
+            isinstance(reg.typ, RegisterType | FloatRegisterType)
+            and reg.typ.data.name is not None
+        )
+
+    def _allocate(self, reg: SSAValue) -> bool:
+        if isinstance(reg.typ, RegisterType | FloatRegisterType) and (
+            available_regs := self.register_sets.get(type(reg.typ))
+        ):
+            if reg.typ.data.name is None:
+                # if we run out of real registers, allocate a j register
+                reg_type = type(reg.typ)
+                if not available_regs:
```

Comment:
Isn't this unreachable? the truthiness check is on line 58

Reply:
Good spot, I simplified this. It needs a test where it runs out of registers